test/metric_value_benchmark/prometheus_test.go

```diff
@@ -0,0 +1,128 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: MIT
+
+//go:build linux && integration
+// +build linux,integration
+
+package metric_value_benchmark
+
+import (
+	_ "embed"
+	"fmt"
+	"log"
+	"time"
+
+	"github.com/aws/amazon-cloudwatch-agent-test/internal/common"
+	"github.com/aws/amazon-cloudwatch-agent-test/test/metric"
+	"github.com/aws/amazon-cloudwatch-agent-test/test/status"
+)
+
+type PrometheusTestRunner struct {
+	BaseTestRunner
+}
+
+var _ ITestRunner = (*PrometheusTestRunner)(nil)
+
+//go:embed agent_configs/prometheus.yaml
+var prometheusConfig string
+
+const prometheusMetrics = `prometheus_test_untyped{include="yes"} 1
+# TYPE prometheus_test_counter counter
+prometheus_test_counter{include="yes"} 1
+# TYPE prometheus_test_counter_exclude counter
+prometheus_test_counter_exclude{include="no"} 1
+# TYPE prometheus_test_gauge gauge
+prometheus_test_gauge{include="yes"} 500
+# TYPE prometheus_test_summary summary
+prometheus_test_summary_sum{include="yes"} 200
+prometheus_test_summary_count{include="yes"} 50
+prometheus_test_summary{include="yes",quantile="0"} 0.1
+prometheus_test_summary{include="yes",quantile="0.5"} 0.25
+prometheus_test_summary{include="yes",quantile="1"} 5.5
+# TYPE prometheus_test_histogram histogram
+prometheus_test_histogram_sum{include="yes"} 300
+prometheus_test_histogram_count{include="yes"} 75
+prometheus_test_histogram_bucket{include="yes",le="0"} 1
+prometheus_test_histogram_bucket{include="yes",le="0.5"} 2
+prometheus_test_histogram_bucket{include="yes",le="2.5"} 3
+prometheus_test_histogram_bucket{include="yes",le="5"} 4
+prometheus_test_histogram_bucket{include="yes",le="+Inf"} 5
+`
+
+func (t *PrometheusTestRunner) validate() status.TestGroupResult {
+	metricsToFetch := t.getMeasuredMetrics()
+	testResults := make([]status.TestResult, len(metricsToFetch))
+	for i, metricName := range metricsToFetch {
+		testResults[i] = t.validatePrometheusMetric(metricName)
+	}
+
+	return status.TestGroupResult{
+		Name:        t.getTestName(),
+		TestResults: testResults,
+	}
+}
+
+func (t *PrometheusTestRunner) getTestName() string {
+	return "Prometheus"
+}
+
+func (t *PrometheusTestRunner) getAgentConfigFileName() string {
+	return "prometheus_config.json"
+}
+
+func (t *PrometheusTestRunner) getAgentRunDuration() time.Duration {
+	return time.Minute
```

Comment:
Question: how does this accurately reflect the run duration?

Reply:
Sorry, didnt get the question. This is to determine how long to run the agent before stopping it to allow a few scrapes. I just kept this the same as others.

Reply:
Oh I just meant does this give you the timestamp, not the duration?

Reply:
Gotcha.. yeah no timestamp.. this just says how long the agent should run for before being stopped